src/components/DatePickerDialog/index.tsx

```diff
@@ -0,0 +1,208 @@
+    placement: 'bottom-start',
+  });
+
+  const closePopper = () => {
+    setIsPopperOpen(false);
+    buttonRef?.current?.focus();
+  };
+
+  const handleInputChange: ChangeEventHandler<HTMLInputElement> = e => {
+    setInputValue(e.currentTarget.value);
+    const date = parse(e.currentTarget.value, DATE_FORMAT, new Date());
+    if (isValid(date)) {
+      setSelected(date);
+    } else {
+      setSelected(undefined);
+      setInputValue(formatDate(new Date())); // if an invalid date string is provided, we default back to the current day
+    }
+  };
+
+  const handleButtonClick = () => {
+    setIsPopperOpen(true);
+  };
+
+  const handleDaySelect = (date: Date | undefined) => {
+    if (!date) return;
+    setSelected(date);
+    if (date) {
+      setInputValue(formatDate(date));
+      closePopper();
+    } else {
+      setInputValue('');
+    }
+  };
+
+  return (
+    <div>
+      <StyledInputContainer ref={popperRef}>
+        <StyledLabel htmlFor={id}>{label}</StyledLabel>
+        <div style={{ width: '100%', display: 'flex', marginTop: '8px' }}>
+          <StyledDateInput
+            id={id}
+            type='text'
+            size={1}
+            value={inputValue}
+            onChange={handleInputChange}
+          />
+          <StyledCalendarButton
+            ref={buttonRef}
+            type='button'
+            aria-label='Wähle ein Datum'
+            onClick={handleButtonClick}
+          >
+            <span aria-hidden={true}>
+              <CalendarToday fontSize='inherit' htmlColor='#37DE8A' />
+            </span>
+          </StyledCalendarButton>
+        </div>
+      </StyledInputContainer>
+      {isPopperOpen && (
+        <FocusTrap
+          active
+          focusTrapOptions={{
+            initialFocus: false,
+            allowOutsideClick: true,
+            clickOutsideDeactivates: true,
+            onDeactivate: closePopper,
+          }}
+        >
+          <div
+            tabIndex={-1}
+            style={popper.styles.popper}
+            className='dialog-sheet'
+            {...popper.attributes.popper}
+            ref={setPopperElement}
+            role='dialog'
+          >
+            <DayPicker
+              locale={de}
+              initialFocus={isPopperOpen}
+              mode='single'
+              defaultMonth={selected}
+              selected={selected}
+              onSelect={handleDaySelect}
+              styles={{
+                root: {
+                  backgroundColor: 'white',
+                  transform: 'none',
+                  border: '1px solid #D9D9D9',
+                  borderRadius: '4px',
+                  margin: 0,
+                  marginTop: '4px',
+                  padding: '10px',
+                },
+                caption: {
+                  color: '#37DE8A',
+                },
+              }}
+              modifiersStyles={{
+                selected: {
+                  backgroundColor: '#37DE8A',
```

Comment:
Should we not use values from the theme here? Hardcoded values seems out of place here. 🫣

Reply:
True. I have refactored this to use the theme now. Btw, I couldn't find a way to get the typed theme from `useTheme`. Let me know if you know how to get the types here.

Reply:
Ok great thx! I can't remember about this. But I think this is not dramatic for now...